File: JavaScript/1164.js
var input = require('fs').readFileSync('/dev/stdin', 'utf8');
var lines = input.split('\n');

var cont = parseInt(lines.shift()); 

for (var i = 0; i < cont; i++) {
  var soma = 0;
  var num = parseInt(lines.shift());
  for (var j = 1; j < num; j++) {
    if (num % j == 0) soma += j;
  }
  if (soma == num) {
    console.log(num + " eh perfeito");
  }
  else {
    console.log(num + " nao eh perfeito");
  }
}

File: Ruby/1164.rb
i = 0
cont = gets.chomp.to_i

while i < cont
    num = gets.chomp.to_i
    j = 1
    s = 0
    while j < num
        if num % j == 0
            s = s + j
        end
        j = j + 1
    end
    
    if s == num
        puts "#{num} eh perfeito"
    else 
        puts "#{num} nao eh perfeito"
    end
    i = i + 1
end